pkg/kv/bulk/sst_batcher.go

```diff
@@ -451,20 +450,51 @@ func (b *SSTBatcher) doFlush(ctx context.Context, reason int) error {
 		b.ms.LastUpdateNanos = timeutil.Now().UnixNano()
 	}
 
-	if err := b.addSSTable(ctx, start, end, b.sstFile.Data()); err != nil {
-		return err
+	stats := b.ms
+	summary := b.rowCounter.BulkOpSummary
+	data := b.sstFile.Data()
+	batchTS := b.batchTS
+	var reserved int64
+
+	fn := func(ctx context.Context) error {
+		if err := b.addSSTable(ctx, batchTS, start, end, data, stats); err != nil {
```

Comment:
sanity check: if we error here it's going to be a fatal error that will cause the flow to tear down (sstbatcher wont be used again) so we don't need to defer a Shrink right?

Reply:
yeah, but we might as well. added.